src/js/components/SocialMediaEditModal.jsx

```diff
@@ -0,0 +1,146 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Form, FormGroup, Label, Input, CardSubtitle, ModalHeader, Modal, ModalBody, ModalFooter } from 'reactstrap';
+import DoWhopButton from './DoWhopButton';
+
+type Props = {
+  handleSocialMediaUrlSubmit: Function,
+  uid: string,
+  socialInputs: Object,
+  handleSocialUrlChange: Function
+};
+type State = {
+  modal: boolean,
+  valueFBValid: boolean,
+  valueTWValid: boolean,
+  valueIGValid: boolean,
+  valueINValid: boolean
+};
+
+class SocialMediaEditModal extends Component<Props, State> {
+  state = {
+    modal: false,
+    valueFBValid: true,
+    valueTWValid: true,
+    valueIGValid: true,
+    valueINValid: true
+  };
+
+  props: Props;
+
+  toggle = () => this.setState(prevState => ({ modal: !prevState.modal }));
+
+  handleChange = (e: Object, site: string) => {
+    this.validateSocialUrl(e, site);
+    this.props.handleSocialUrlChange(e, site);
+  };
+
+  handleSubmit = () => {
+    const socialUrls = {
+      facebookUrl: this.props.socialInputs.valueFB,
+      twitterUrl: this.props.socialInputs.valueTW,
+      instagramUrl: this.props.socialInputs.valueIG,
+      linkedInUrl: this.props.socialInputs.valueIN
+    };
+    this.props.handleSocialMediaUrlSubmit(socialUrls, this.props.uid);
+    this.toggle();
+  };
+
+  validateSocialUrl = (e: Object, site: string) => {
+    const urlRegEx = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/;
```

Comment:
Perhaps here is where we could do extra 'cleaning'

Reply:
Yes! This is the spot to ad a fancy regexs to only allow profiles, it has an added complication of needing one for each social media type.